tools/github-actions/create-branch-e2e-screenshots/action.yml

```
@@ -0,0 +1,68 @@
+name: Create branch e2e screenshots
+description: Create a branch with the update of the e2e screenshots
+inputs:
+  reportPath:
+    type: string
+    description: 'Path to the custom report'
+    required: true
+  screenshotsPattern:
+    type: string
+    description: 'Pattern to detect which screenshots should be committed'
+    required: true
+  branchSuffix:
+    type: string
+    description: 'Suffix for the branch that will be created'
+    required: false
+    default: '-update-e2e-screenshots'
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Update e2e screenshots
+      id: update-e2e-screenshots
+      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      env:
+        REPORT_PATH: ${{ inputs.reportPath }}
+      with:
+        script: |
+          const screenshots = JSON.parse(require('node:fs').readFileSync(process.env.REPORT_PATH, 'utf8'));
+          screenshots.forEach((screenshot) => require('node:child_process').spawnSync('cp', [screenshot.actual, screenshot.expected]));
+          core.setOutput('screenshots', screenshots.length);
+    - name: Define shared variables
+      if: steps.update-e2e-screenshots.outputs.screenshots != '0'
+      id: vars
+      env:
+        # run_id and run_attempt are needed to avoid conflict in the branch name
+        BRANCH_NAME: ${{ github.head_ref }}${{ inputs.branchSuffix }}-${{ github.run_id }}-${{ github.run_attempt }}
+      shell: bash
+      run: |
+        echo "BRANCH_NAME=$BRANCH_NAME" >> $GITHUB_OUTPUT
+    - name: Create branch
+      id: auto-commit-action
+      if: steps.update-e2e-screenshots.outputs.screenshots != '0'
+      uses: stefanzweifel/git-auto-commit-action@e348103e9026cc0eee72ae06630dbe30c8bf7a79 # v5.1.0
+      with:
+        commit_message: "fix: update e2e screenshots"
```

```suggestion
        commit_message: "test: update e2e screenshots"
```
or `chore` since it's not actually a code fix.

Also, can you add something to the commit not to have the same message everytime (like the input branch)

---

It's a temporary commit 
you have to cherry-pick it in your PR and squash it with your original commit

---

but you are not forced to squash it :-s

---

We are used to accept PR with only 1 commit